tasks/insertionTasks.ts

```diff
@@ -0,0 +1,235 @@
+
+        console.log(`Update needed: ${currentSHA}..${options.roslynEndSHA}`);
+
+        // Step 4: Verify Roslyn repo exists
+        if (!options.roslynRepoPath) {
+            throw new Error('roslynRepoPath is required');
+        }
+        await verifyRoslynRepo(options.roslynRepoPath);
+
+        // Step 5: Generate PR list
+        const prList = await generatePRList(currentSHA, options.roslynEndSHA, options.roslynRepoPath, options);
+        console.log('PR List generated:', prList);
+
+        // Check if PR list is empty or contains no meaningful PRs
+        if (!prList || prList === '(failed to generate PR list, see pipeline for details)') {
+            console.log('No PRs with required labels found. Skipping insertion.');
+            logWarning('No PRs with VSCode label found between the commits. Skipping insertion.');
+            return;
+        }
+
+        // Step 6: Update files
+        await updatePackageJson(options.roslynVersion);
+        await updateChangelog(options.roslynVersion, prList, options.roslynBuildNumber, options.roslynBuildId);
+
+        // Step 7: Create branch and PR
+        const prTitle = `Bump Roslyn to ${options.roslynVersion} (${options.roslynEndSHA?.substring(0, 8)})`;
+        const prBody = `This PR updates Roslyn to version ${options.roslynVersion} (${options.roslynEndSha}).\n\n${prList}`;
+        const commitMessage = `Bump Roslyn to ${options.roslynVersion} (${options.roslynEndSha?.substring(0, 8)})`;
+
+        await createBranchAndPR({
+            ...options,
+            commitSha: options.roslynEndSHA!,
+            targetRemoteRepo: 'vscode-csharp',
+            baseBranch: options.targetBranch || 'main',
+            newBranchName: `insertion/${options.roslynEndSHA}`,
+            githubPAT: options.githubPAT!,
+            dryRun: options.dryRun,
+            userName: options.userName,
+            email: options.email
+        }, prTitle, commitMessage, prBody);
+
+    } catch (error) {
+        const errorMessage = error instanceof Error ? error.message : String(error);
+        logError(`Insertion failed: ${errorMessage}`);
+        if (error instanceof Error && error.stack) {
+            console.log(`##[debug]${error.stack}`);
+        }
+        throw error;
+    }
+});
+
+async function extractRoslynVersionFromManifest(manifestPath: string): Promise<string | null> {
+    const xmlFile = path.join(manifestPath, 'OfficialBuild.xml');
+
+    if (!fs.existsSync(xmlFile)) {
+        logError(`OfficialBuild.xml not found at ${xmlFile}`);
+        return null;
+    }
+
+    const xmlContent = fs.readFileSync(xmlFile, 'utf8');
+    const parser = new xml2js.Parser();
+    const result = await parser.parseStringPromise(xmlContent);
+
+    const packages = result?.Build?.Package || [];
+    for (const pkg of packages) {
+        const attrs = pkg.$;
+        if (attrs?.Id === 'Microsoft.CodeAnalysis.Common') {
+            return attrs.Version;
+        }
+    }
+
+    return null;
```

Comment:
Log an Error that the Microsoft.CodeAnalysis.Common package could not be found. That would be quite a problem.